src/utilities/buildASTSchema.js

```diff
@@ -132,10 +132,9 @@ function getNamedTypeNode(typeNode: TypeNode): NamedTypeNode {
  * If no schema definition is provided, then it will look for types named Query
  * and Mutation.
  *
- * Given that AST it constructs a GraphQLSchema. The resulting schema
- * has no resolve methods, so execution will use default resolvers.
+ * Given that AST it return object that can be used to construct GraphQLSchema.
  */
-export function buildASTSchema(ast: DocumentNode): GraphQLSchema {
+export function buildASTSchemaConfig(ast: DocumentNode) {
```

Comment:
Yeah, sorry: I didn't mean literally `GraphQLSchemaConfig`; I meant some name like it.

In the worst case you define two types with different names. Better though, would be to use an intersection type:

```
type Base = {
  mutation?: ?GraphQLObjectType;
  subscription?: ?GraphQLObjectType;
  types?: ?Array<GraphQLNamedType>;
  directives?: ?Array<GraphQLDirective>;
};

type A = Base & {
  query: GraphQLObjectType;
};

type B = Base & {
  query?: GraphQLObjectType;
};
```

(Leaving the actual names of `A` and `B` out just for illustration.)